src/lib/u2.js

```diff
@@ -47,6 +47,59 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'CSS',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'code'
+    },
+    {
+      name: 'name',
+      factory: function() { return 'CSS-' + this.$UID; }
+    },
+    {
+      name: 'installedDocuments_',
+      factory: function() {
+        return new WeakMap();
+      }
+    }
+  ],
+
+  methods: [
+    function installInClass(cls) {
+      // Install myself in this Window, if not already there.
+      var oldCreate = cls.create;
+      var axiom = this;
+
+      cls.create = function(args, opt_parent) {
+        // TODO: move this functionality somewhere reusable
+        var X = opt_parent ?
+          (opt_parent.__subContext__ || opt_parent.__context__ || opt_parent) :
+          foam.__context__;
+
+        // Install our own CSS, and then all parent models as well.
+        if ( ! axiom.installedDocuments_.has(X.document) ) {
+          X.installCSS(axiom.expandCSS(cls, axiom.code));
+          axiom.installedDocuments_.set(X.document, true);
+        }
+
+        // Now call through to the original create.
+        return oldCreate.call(this, args, X);
+      };
+    },
+
+    function expandCSS(cls, text) {
+      /* Performs expansion of the ^ shorthand on the CSS. */
+      // TODO(braden): Parse and validate the CSS.
+      // TODO(braden): Add the automatic prefixing once we have the parser.
+      return text.replace(/\^/g,
```

Comment:
This is how it worked in FOAM1 U2. It's hard to tell if a prefix is following, especially in the parser-integrated logic for this in FOAM1.

I can try to avoid the trailing `-` here if you think it's worthwhile.